web/src/client/components/multiplexer-modal.ts

```diff
@@ -0,0 +1,891 @@
+import type { PropertyValues } from 'lit';
+import { css, html, LitElement } from 'lit';
+import { customElement, property, state } from 'lit/decorators.js';
+import { repeat } from 'lit/directives/repeat.js';
+import type {
+  MultiplexerStatus,
+  MultiplexerTarget,
+  MultiplexerType,
+  TmuxPane,
+  TmuxWindow,
+} from '../../shared/multiplexer-types.js';
+import { apiClient } from '../services/api-client.js';
+import { Z_INDEX } from '../utils/constants.js';
+import './modal-wrapper.js';
+
+@customElement('multiplexer-modal')
+export class MultiplexerModal extends LitElement {
+  static styles = css`
+    :host {
+      position: fixed;
+      inset: 0;
+      z-index: ${Z_INDEX.MODAL};
+      display: none;
+      align-items: center;
+      justify-content: center;
+      padding: 1rem;
+    }
+
+    :host([open]) {
+      display: flex;
+    }
+
+    .content {
```

Comment:
if you instruct the agent to disable shadow-dom here and use tailwind, we can get rid of a lot of custom css.